lib/src/command/lish.dart

```diff
@@ -67,7 +68,8 @@ class LishCommand extends PubCommand {
         return log.progress('Uploading', () async {
           // TODO(nweiz): Cloud Storage can provide an XML-formatted error. We
           // should report that error and exit.
-          var newUri = server.resolve("/api/packages/versions/new");
+          var newUri = server.replace(
```

Comment:
Would this work?:
```
var newUri = server.resolve("api/packages/versions/new");
```

Reply:
If your base URL is `example.com/sub/dir` then you end up with `example.com/sub/api/packages/versions/new`, I guess because `api` is technically a “file” path and not a “directory” path component. I *didn’t* test it with `example.com/sub/dir/`, though, but also, I think that if users were required to include the trailing slash in `$PUB_HOSTED_URL`, many people would be confused.

Reply:
>I think that if users were required to include the trailing slash in $PUB_HOSTED_URL, many people would be confused.

I agree

We have another strategy at: https://github.com/dart-lang/pub/blob/master/lib/src/source/hosted.dart#L316 , just doing sting concatenation. It somehow 'feels' wrong, but seems to work.

I would prefer using just Uri methods, and not package:path.

@jonasfj do you have any ideas?

Reply:
@sigurdm, I think not using `package:path` is nice.. but maybe we should make it all consistent and just land it?